webapp/sources/rudder/rudder-web/src/main/elm/rules/sources/View.elm

```diff
@@ -35,27 +36,33 @@ view model =
           ]
         ]
 
-    ruleTreeCategory : (Category Rule) -> Html Msg
+    ruleTreeCategory : (Category Rule) -> Maybe (Html Msg)
     ruleTreeCategory item =
       let
         categories = getSubElems item
-                       |> List.sortBy .name
-                       |> List.map ruleTreeCategory
+          |> List.sortBy .name
+          |> List.filterMap ruleTreeCategory
+
         rules = item.elems
-                |> List.sortBy .name
-                |> List.map ruleTreeElem
+          |> List.filter (filterRules model)
+          |> List.sortBy .name
+          |> List.map ruleTreeElem
 
         childsList  = ul[class "jstree-children"] (List.concat [ categories, rules] )
-
       in
-        li[class "jstree-node jstree-open"]
-        [ i[class "jstree-icon jstree-ocl"][]
-        , a[href "#", class "jstree-anchor", onClick (OpenCategoryDetails item)]
-          [ i [class "jstree-icon jstree-themeicon fa fa-folder jstree-themeicon-custom"][]
-          , span [class "treeGroupCategoryName tooltipable"][text item.name]
-          ]
-        , childsList
-        ]
+        if (String.isEmpty model.ui.ruleFilters.filter) || ((List.length rules > 0) || (List.length categories > 0)) then
```

Comment:
Don't return a maybe here 

do 

`if (String.isEmpty model.ui.ruleFilters.filter) || ((List.length rules > 0) || (List.length categories > 0)) then
            li[class "jstree-node jstree-open"]
          [ i [class "jstree-icon jstree-themeicon fa fa-folder jstree-themeicon-custom"][]
            [ i[class "jstree-icon jstree-ocl"][]
          , span [class "treeGroupCategoryName tooltipable"][text item.name]
            , a[href "#", class "jstree-anchor", onClick (OpenCategoryDetails item)]
          ]
              [ i [class "jstree-icon jstree-themeicon fa fa-folder jstree-themeicon-custom"][]
        , childsList
              , span [class "treeGroupCategoryName tooltipable"][text item.name]
        ]
              ]
            , childsList
            ]
else 
  div [class "alert alert-warning"]
                  [ i [class "fa fa-exclamation-triangle"][]
                  , text  "No rules match your filter."
                  ]
 `

Reply:
with correct indent, indeed but github cannot do it or at least i can't do it :)

Reply:
I have to, because List.filterMap needs the first argument to be a Maybe :
<pre> |> List.filterMap ruleTreeCategory</pre>

Thanks to this, I can hide categories that do not contain any rules that match the filters.